src/app/external-cluster-wizard/steps/external-cluster/provider/aks/component.ts

```diff
@@ -318,7 +344,7 @@ export class AKSClusterSettingsComponent
             name: this.controlValue(Controls.NodePoolName),
             basicSettings: {
               mode: this.controlValue(Controls.Mode),
-              vmSize: this.controlValue(Controls.VmSize)?.main,
+              vmSize: this.controlValue(Controls.VmSize)[ComboboxControls.Select],
```

Comment:
To be on safe side. Please also add below.
```suggestion
              vmSize: this.controlValue(Controls.VmSize)?.[ComboboxControls.Select],
```

Reply:
This is still missing